src/snoowrap.js

```diff
@@ -639,6 +639,45 @@ const snoowrap = class snoowrap {
     return this._get({uri: 'api/saved_categories'}).get('categories');
   }
 
+  /**
+   * @param {string} query a query to search
+   * @param {Object} [config] config object
+   * @returns {Promise} A promise with array os subreddits starting with that query
+   * @example
+   * const config = {
+   *  exact:false,
+   *  include_over_18:false,
+   *  include_unadvertisable:false
+   * };
+   * const {subreddits} = await r.searchSubreddit("learn", config);
+   * => [
+   * {
+   *   active_user_count: 1705
+   *   allow_chat_post_creation: null
+   *   allow_images: false
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnprogramming"
+   *   subscriber_count: 1084731
+   * },
+   * {
+   *   active_user_count: 126
+   *   allow_chat_post_creation: null
+   *   allow_images: true
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnjavascript"
+   *   subscriber_count: 76606
```

Comment:
This should depict a Listing of Subreddits ideally, similar to how `getHot()` example shows https://github.com/not-an-aardvark/snoowrap/blob/a28870bfbb83448b54b438a3a37c29ee60b6f502/src/snoowrap.js#L817-L822

Reply:
But it's not actually return array of instances of `Subreddit`. It simply return an array of objects with the specified props. `getHot()` is different, it returns array of instances that the library has built right?

Reply:
Ah I see. Could probably wrap it to a Listing of Subreddits manually through `_getListing()` and use `_transform` to convert it into unfetched Subreddit objects https://github.com/not-an-aardvark/snoowrap/blob/9d412777826f001fb33ea287b7b80d77af4d1862/src/snoowrap.js#L1084-L1106

Reply:
Ah right I see. You should be able to utilize `_getListing()` then and use its `_transform` argument to convert it into unfetched Subreddit objects

Reply:
So I have ton of questions about the `_transform` function and how to use it but first - it says the function is kind of a `_get` function, but the endpoint require it to be `_post`. How can I handle it using `_getListing`?

Reply:
It would be similar to how `getNewModmailConversations()` works

https://github.com/not-an-aardvark/snoowrap/blob/9d412777826f001fb33ea287b7b80d77af4d1862/src/snoowrap.js#L1084-L1106

`_transform` gives the response as is so you can transform it how you want. Simply map the subreddits from the response into Subreddit objects then wrap that in a Listing

Reply:
ok cool. How about that the `_getListing` is a `GET` request and the endpoint expect for `POST`?

Reply:
Ah right.. I'll have to think about it unless @not-an-aardvark has thoughts. Otherwise, probably creating a `postListing()` could be done - though not sure about that name since it would be confusing

Reply:
How important is to convert it to `Listing` of `Subreddit`? is it just for consistency? I mean what's the upside here?

Reply:
Listing is just what we use everywhere so it's good to use that when we have an array of object. 

For mapping to a Subreddit object is simply because we get enough data to construct objects from the response. With it we can also chain more requests to do actions or retrieve data

Reply:
I think just using `_post` and transforming inside `then()` should work. We can create a listing there similarly to how `_getListing()` is done in `getNewModmailConversations()`